pkg/storegateway/bucket_e2e_test.go

```diff
@@ -506,6 +521,69 @@ func TestBucketStore_e2e(t *testing.T) {
 	})
 }
 
+func TestBucketStore_e2e_StreamingEdgeCases(t *testing.T) {
+	foreachStore(t, func(t *testing.T, newSuite suiteFactory) {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		s := newSuite(func(config *prepareStoreConfig) {
+			config.shiftedBlocks = true
+		})
+
+		_, maxt := s.store.TimeRange()
+		additionalCases := []testBucketStoreCase{
+			{ // This tests if the first phase of streaming that sends only the series is filtering the series by chunk time range.
+				// The request time range overlaps with 2 blocks with 4 timeseries each, but only the 2nd block
+				// has some overlapping data that should be returned.
+				req: &storepb.SeriesRequest{
+					Matchers: []storepb.LabelMatcher{
+						{Type: storepb.LabelMatcher_RE, Name: "a", Value: "1|2"},
+					},
+					MinTime: maxt - 121*int64(time.Minute/time.Millisecond),
```

Comment:
Where does 121 come from?

Reply:
120 mins is how much a block spans. So 121 so that we exclude the last block.

Reply:
I see it now. It is 121 so that we include the N-1'th block. I will add that comment,